src/metadata-builder/RelationJoinColumnBuilder.ts

```diff
@@ -233,6 +233,11 @@ export class RelationJoinColumnBuilder {
                     },
                 })
                 relation.entityMetadata.registerColumn(relationalColumn)
+            } else if (relationalColumn.referencedColumn) {
+                relationalColumn = Object.assign(
+                    Object.create(ColumnMetadata.prototype) as ColumnMetadata,
```

Comment:
isn't this the same as `new ColumnMetadata`?

Reply:
I'm not sure, but using the class constructor it is possible to create an exact clone.

In the PR description
> In RelationJoinColumnBuilder.collectColumns, relationalColumn is used by reference and referencedColumn is overwritten. Solution: Create a copy of the found relationalColumn.

Please recommend what comment text to leave in the code